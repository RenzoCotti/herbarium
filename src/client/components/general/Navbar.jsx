import React, { Component } from "react";
import { NavLink } from "react-router-dom";

class Navbar extends Component {
  constructor(props) {
    super(props);
    this.getOptions = this.getOptions.bind(this);
  }

  state = {
    options: [
      "Home",
      "Identify",
      "Browse",
      "Random",
      "Create",
      "Search",
      "About"
    ]
  };

  getOptions() {
    return (
      <React.Fragment>
        {this.state.options.map(op => {
          let link = op.toLowerCase();
          if (op === "Home") link = "";
          return (
            <NavLink className="nav-element" key={op} to={"/" + link}>
              {op}
            </NavLink>
          );
        })}
      </React.Fragment>
    );
  }

  render() {
    return <div className="navbar">{this.getOptions()}</div>;
  }
}

export default Navbar;
