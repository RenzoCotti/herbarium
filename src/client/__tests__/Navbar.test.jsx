import Navbar from "../components/Navbar";
import React from "react";
import { shallow } from "../enzyme";

describe("Navbar", () => {
  it("renders correctly", () => {
    const nav = shallow(<Navbar />);
    expect(nav).toMatchSnapshot();
  });
  it("should have 6 children", () => {
    const nav = shallow(<Navbar />);
    expect(nav.children()).toHaveLength(6);
  });
  it("should have the correct labels", () => {
    const nav = shallow(<Navbar />);
    let arr = ["Home", "Identify", "Browse", "Random", "Search", "About"];

    nav.children().forEach(function(child, i) {
      expect(child.text()).toBe(arr[i]);
    });
  });
});
