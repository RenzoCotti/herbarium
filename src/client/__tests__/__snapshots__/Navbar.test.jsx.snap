// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <React.Fragment>
        <div
          className="nav-element"
        >
          Home
        </div>
        <div
          className="nav-element"
        >
          Identify
        </div>
        <div
          className="nav-element"
        >
          Browse
        </div>
        <div
          className="nav-element"
        >
          Random
        </div>
        <div
          className="nav-element"
        >
          Search
        </div>
        <div
          className="nav-element"
        >
          About
        </div>
      </React.Fragment>,
      "className": "navbar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="nav-element"
          >
            Home
          </div>,
          <div
            className="nav-element"
          >
            Identify
          </div>,
          <div
            className="nav-element"
          >
            Browse
          </div>,
          <div
            className="nav-element"
          >
            Random
          </div>,
          <div
            className="nav-element"
          >
            Search
          </div>,
          <div
            className="nav-element"
          >
            About
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Home",
          "nodeType": "host",
          "props": Object {
            "children": "Home",
            "className": "nav-element",
          },
          "ref": null,
          "rendered": "Home",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Identify",
          "nodeType": "host",
          "props": Object {
            "children": "Identify",
            "className": "nav-element",
          },
          "ref": null,
          "rendered": "Identify",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Browse",
          "nodeType": "host",
          "props": Object {
            "children": "Browse",
            "className": "nav-element",
          },
          "ref": null,
          "rendered": "Browse",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Random",
          "nodeType": "host",
          "props": Object {
            "children": "Random",
            "className": "nav-element",
          },
          "ref": null,
          "rendered": "Random",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Search",
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "className": "nav-element",
          },
          "ref": null,
          "rendered": "Search",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "About",
          "nodeType": "host",
          "props": Object {
            "children": "About",
            "className": "nav-element",
          },
          "ref": null,
          "rendered": "About",
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <React.Fragment>
          <div
            className="nav-element"
          >
            Home
          </div>
          <div
            className="nav-element"
          >
            Identify
          </div>
          <div
            className="nav-element"
          >
            Browse
          </div>
          <div
            className="nav-element"
          >
            Random
          </div>
          <div
            className="nav-element"
          >
            Search
          </div>
          <div
            className="nav-element"
          >
            About
          </div>
        </React.Fragment>,
        "className": "navbar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="nav-element"
            >
              Home
            </div>,
            <div
              className="nav-element"
            >
              Identify
            </div>,
            <div
              className="nav-element"
            >
              Browse
            </div>,
            <div
              className="nav-element"
            >
              Random
            </div>,
            <div
              className="nav-element"
            >
              Search
            </div>,
            <div
              className="nav-element"
            >
              About
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Home",
            "nodeType": "host",
            "props": Object {
              "children": "Home",
              "className": "nav-element",
            },
            "ref": null,
            "rendered": "Home",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "Identify",
            "nodeType": "host",
            "props": Object {
              "children": "Identify",
              "className": "nav-element",
            },
            "ref": null,
            "rendered": "Identify",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "Browse",
            "nodeType": "host",
            "props": Object {
              "children": "Browse",
              "className": "nav-element",
            },
            "ref": null,
            "rendered": "Browse",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "Random",
            "nodeType": "host",
            "props": Object {
              "children": "Random",
              "className": "nav-element",
            },
            "ref": null,
            "rendered": "Random",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "Search",
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "className": "nav-element",
            },
            "ref": null,
            "rendered": "Search",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "About",
            "nodeType": "host",
            "props": Object {
              "children": "About",
              "className": "nav-element",
            },
            "ref": null,
            "rendered": "About",
            "type": "div",
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
